torch/csrc/distributed/c10d/init.cpp

```diff
@@ -650,11 +650,13 @@ Example::
           .def(
               "get",
               [](::c10d::Store& store, const std::string& key) -> py::bytes {
-                auto value = store.get(key);
+                auto value = [&]() {
+                  py::gil_scoped_release guard;
```

Comment:
Can you do what's being done on L627 with `py::call_guard<py::gil_scoped_release>(),` outside of the function implementation?

Reply:
It is fine to use as long as you don't call into python API inside the lambda.
The problem here is only the call to `py::bytes()` as this creates a new python object and thus must have the GIL held.
This is why the fix here is to restrict the guard to only apply to your c++ function and not to the py::bytes call.